.github/workflows/auto-publish.yml

```
@@ -0,0 +1,74 @@
+name: Publish stytch PyPI package
+
+on:
+  push:
+    branches: [main]
+
+jobs:
+  build-and-publish:
+    runs-on: ubuntu-latest
+    name: Build and publish stytch package to PyPI
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.11"
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+
+      - name: Build source tarball
+        run: python setup.py sdist
+
+      - name: Get version
+        id: version
+        run: |
+          VERSION=$(python -c 'import stytch; print(stytch.__version__)')
+          echo "version=$VERSION" >> $GITHUB_OUTPUT
+          echo "release_tag=v$VERSION" >> $GITHUB_OUTPUT
+
+      - name: Get changed files
+        id: files
+        uses: jitterbit/get-changed-files@v1
+
+      - name: Check for version.py diff
+        id: diff
+        run: |
+          FOUND=0
+          for changed_file in ${{ steps.files.outputs.all }}; do
+            if [[ $changed_file == "stytch/version.py" ]]; then
+              FOUND=1
+            fi
+          done
+          echo "diff=$FOUND" >> $GITHUB_OUTPUT
```

This works and makes sense to me. But is this worth the dependency over something like this for a single file?
```suggestion
      - name: Check for version.py diff
        id: diff
        run: |
          if git diff --exit-code --quiet "$GITHUB_BASE_REF" -- stytch/version.py; then
            echo "diff=true" >> $GITHUB_OUTPUT
          else
            echo "diff=false" >> $GITHUB_OUTPUT
          fi
```

---

I tried something similar but it wasn't working because by default `actions/checkout` doesn't fetch enough information (and even with `fetch-depth: 0`, I was getting cryptic errors). Let me try this, though.

---

Yeah, even with `fetch-depth: 10`, I'm getting weird stuff here.
https://github.com/logan-stytch/stytch-python/actions/runs/3745882995/jobs/6360740782

I'm inclined to just land this for now, but I agree that it should be way easier. My initial attempt just did `git diff --name-only HEAD^ HEAD | grep stytch/version.py` but it failed with an even worse error: `Process completed with exit code 1.`
Here's a sample run for that one: https://github.com/logan-stytch/stytch-python/actions/runs/3743303647

---

Weird! Maybe I have the GitHub env var wrong (should've cited instead of going from memory), but this at least convinces me that it's non-obvious and that the third-party action is worth it. Thanks for entertaining the suggestion!

---

BACK-2331